src/test/java/uk/gov/hmcts/dts/fact/services/CourtServiceTest.java

```diff
@@ -472,19 +476,68 @@ void shouldReturnListForNearestCourtsByPostcodeSearch() {
         when(serviceArea.getSlug()).thenReturn(serviceAreaSlug);
         when(serviceAreaRepository.findBySlugIgnoreCase(serviceAreaSlug)).thenReturn(Optional.of(serviceArea));
         when(mapitService.getMapitData(any())).thenReturn(Optional.of(mapitData));
-        when(serviceAreaSearchFactory.getSearchFor(serviceArea, mapitData)).thenReturn(search);
+        when(serviceAreaSearchFactory.getSearchFor(serviceArea, mapitData, Action.NOT_LISTED)).thenReturn(search);
         when(search.searchWith(serviceArea, mapitData, JE2_4BA)).thenReturn(courts);
 
         final ServiceAreaWithCourtReferencesWithDistance results = courtService.getNearestCourtsByPostcodeSearch(
             JE2_4BA,
-            serviceAreaSlug
+            serviceAreaSlug,
+            Action.NOT_LISTED
+        );
+
+        assertThat(results.getSlug()).isEqualTo(serviceAreaSlug);
+        assertThat(results.getCourts().size()).isEqualTo(2);
+        assertThat(results.getCourts().get(0)).isInstanceOf(CourtReferenceWithDistance.class);
+    }
+
+    @Test
+    void shouldReturnListForNearestCourtsByPostcodeActionAndAreaOfLawSearch() {
+
+        final String serviceAreaSlug = "childcare-arrangements";
+        final String postcode = "RM19 1SR";
+        final ServiceArea serviceArea = mock(ServiceArea.class);
```

Comment:
Up to you on this one, I see that other tests here have the same duplicated code

You could move these to the top of the class, and annotate them with a tag: 

I.e 

@mock 
private final ServiceArea serviceAreaMock;